src/SourceGenerators/Uno.UI.SourceGenerators/Helpers/HashBuilder.cs

```diff
@@ -15,8 +15,8 @@ internal class HashBuilder
 		/// <summary>
 		/// Build a stable ID from the provided symbol
 		/// </summary>
-		public static string BuildIDFromSymbol(ITypeSymbol typeSymbol)
-			=> typeSymbol.GetFullName() + "_" + Build(typeSymbol.MetadataName + typeSymbol.ContainingAssembly.MetadataName);
+		public static string BuildIDFromSymbol(INamedTypeSymbol typeSymbol)
+			=> typeSymbol.GetFullMetadataNameForFileName() + "_" + Build(typeSymbol.MetadataName + typeSymbol.ContainingAssembly.MetadataName);
```

Comment:
`GetFullName` is too slow. Other callsites should possibly be reviewed for performance improvements.